cmake/compiler_support.cmake

```diff
@@ -31,22 +31,65 @@ if (NOT PERL_FOUND )
    message(FATAL_ERROR "Perl not found")
 endif()
 
+
 set(MASM2GAS_PATH ${OMR_ROOT}/tools/compiler/scripts/masm2gas.pl CACHE INTERNAL "MASM2GAS PATH")
 
 include(${OMR_ROOT}/cmake/AddPrefix.cmake) 
 
-# Platform setup code!
-# TODO: THis needs to be abstracted for cross platform builds
+# Fetch the OMR view of the system.
+include(OmrDetectSystemInformation)
+
+macro(tr_detect_system_information)
+
+	macro(jit_not_ready)
+		message(FATAL "JIT isn't ready to build with CMake on this platform")
+	endmacro()
+
+
+	omr_detect_system_information()
+
+
+	set(TR_COMPILE_DEFINITIONS "")
+
+
+	# Platform setup code! 
+	# TODOs:
+	#  - Support more platforms, and, separate host and target arch. 
+	#  - Once we support all the platforms OMR supports with CMake, this can be
+	#    largely integrated into OmrDetectSystemInformation.cmake. 
+	if(OMR_ARCH_X86)
+		set(TR_HOST_ARCH    x     CACHE INTERNAL "The architecture directory used for the compiler code (x, p, or z)")
```

Comment:
So, I would personally defend alignment: You read the code more than you read diffs, and if the alignment make it easier to read the code (as I think it does here), I'd put that higher in importance than minimizing diffs.

Having said that, this patch is known to not be final form. I'm trying to get something in here that will allow #1475 to go in, but the CMake platform story us under active evolution.